zerver/tests/test_events.py

```diff
@@ -972,70 +973,54 @@ def test_change_is_admin(self):
         else:
             raise AssertionError("Unexpected property type %s" % (property_type,))
+        schema_checker = self.check_events_dict([
+            ('type', equals('update_display_settings')),
+            ('setting_name', equals(setting_name)),
+            ('user', check_string),
+            ('setting', validator),
+        ])
 
-        num_events = 1
-        if setting_name == "timezone":
-            num_events = 2
-        if property_type == bool:
-            do_set_user_display_setting(self.user_profile, setting_name, False)
-        for value in values_list:
-            events = self.do_test(lambda: do_set_user_display_setting(
-                self.user_profile, setting_name, value), num_events=num_events)
-
-            schema_checker = self.check_events_dict([
-                ('type', equals('update_display_settings')),
-                ('setting_name', equals(setting_name)),
-                ('user', check_string),
-                ('setting', validator),
-            ])
-            error = schema_checker('events[0]', events[0])
-            self.assert_on_error(error)
-
-            timezone_schema_checker = self.check_events_dict([
-                ('type', equals('realm_user')),
-                ('op', equals('update')),
-                ('person', check_dict_only([
-                    ('email', check_string),
-                    ('user_id', check_int),
-                    ('timezone', check_string),
-                ])),
-            ])
-            if setting_name == "timezone":
-                error = timezone_schema_checker('events[1]', events[1])
-
-    def test_change_twenty_four_hour_time(self):
-        # type: () -> None
-        self.do_set_user_display_settings_test("twenty_four_hour_time", [True, False])
-
-    def test_change_left_side_userlist(self):
-        # type: () -> None
-        self.do_set_user_display_settings_test("left_side_userlist", [True, False])
-
-    def test_change_emoji_alt_code(self):
-        # type: () -> None
-        self.do_set_user_display_settings_test("emoji_alt_code", [True, False])
-
-    def test_change_emojiset(self):
-        # type: () -> None
-        self.do_set_user_display_settings_test("emojiset", [u'apple', u'twitter'])
+        changes = test_changes.get(setting_name)
+        if (changes is None):
+            raise AssertionError('No test created for %s' % (setting_name))
+        do_set_user_display_setting(self.user_profile, setting_name, changes[0])
+        for change in changes[1:]:
+            if setting_name == 'timezone':
+                events = self.do_test(lambda: do_set_user_display_setting(self.user_profile, setting_name, change), num_events=2)
+            else:
+                events = self.do_test(
+                    lambda: do_set_user_display_setting(self.user_profile, setting_name, change))
+                error = schema_checker('events[0]', events[0])
+                self.assert_on_error(error)
```

Comment:
I think I preferred the old arrangement of the code of this bottom section, that used the `num_events` variable, since it was more readable.